### PR TITLE
Adding clone and copy traits to App

### PR DESCRIPTION
**Pull request by ChrisB9**
I'd like to be able to loop through a list and create the commands from that. This change makes it very comfortable to do so

- [x]  Added Tests

### CODE REVIEW

File: src/app.rs [hunk 1]
```diff
@@ -623,4 +623,14 @@ mod tests {
...
     }
+
+    #[test]
+    fn create_commands_through_for_loop() {
+        let app = App::new("test").flag(Flag::new("bool", FlagType::Bool));
+        vec!["dev", "prod"]
+            .into_iter()
+            .map(|a| app.clone().command(Command::new(a)));
+        app.run(vec!["test".to_string(), "prod".to_string()]);
+        assert_eq!(app.name, "test".to_string());
```
Comment (ksk001100) [inline src/app.rs:634]: @ChrisB9 

Thanks for the PR !
Please fix the test like this.

```suggestion
       let apps = vec!["dev", "prod"]
            .into_iter()
            .map(|a| app.clone().command(Command::new(a)));
        
        for app in apps {
            app.run(vec!["test".to_string(), "prod".to_string()]);
            assert_eq!(app.name, "test".to_string());
        }
```

**Comment by ChrisB9** [comment]
Thanks for the suggestion!

One thing I am not sure about, is, how I can enable the copy trait on it, do you have an idea about that?

**Comment by rnitta** [comment]
@ChrisB9 
> I'd like to be able to loop through a list and create the commands from that.

Sorry but I don't understand what it means.
Would you mind suggesting the specific usecase which is enabled by this PR?

**Comment by ChrisB9** [comment]
> Would you mind suggesting the specific usecase which is enabled by this PR?

Sure, I have a list of subcommands preconfigured in a toml file. I want to iterate over this list and add each subcommand to `App`.
In pseudo-code:

```rust
let app = App::new();
let toml_list_commands = commands_from_toml();

for cmd in toml_list_commands {
   app.command(cmd);
}

app.run();
```


This usecase is not yet fully enabled by this PR, but I want to get there

**Comment by ksk001100** [comment]
@ChrisB9 
For that use case, you can solve the problem by using fold

```rust
fn main() {
    let args: Vec<String> = env::args().collect();
    let app = App::new("test");
    let toml_list_commands = commands_from_toml();

    let app = toml_list_commands.into_iter().fold(app, |app, cmd| {
        app.command(cmd)
    });

    app.run(args);
}
```

**Comment by ChrisB9** [comment]
@ksk001100 
Thats a really nice solution, thank you!

**Comment by ksk001100** [comment]
This change is no longer needed and will be closed.

**Pull request closed without merging**